aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -1,35 +1,50 @@
 package org.wordpress.aztec.spans
 
 import android.content.Context
+import android.graphics.Bitmap
 import android.graphics.Canvas
 import android.graphics.Paint
 import android.graphics.Rect
+import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.Drawable
 import android.view.Gravity
 import org.wordpress.aztec.AztecAttributes
 import org.wordpress.aztec.AztecText
 import java.util.*
 
-abstract class AztecMediaSpan(context: Context, drawable: Drawable?, override var attributes: AztecAttributes = AztecAttributes(),
+
+abstract class AztecMediaSpan(context: Context, imageProvider: IImageProvider, override var attributes: AztecAttributes = AztecAttributes(),
                               var onMediaDeletedListener: AztecText.OnMediaDeletedListener? = null,
-                              editor: AztecText? = null) : AztecDynamicImageSpan(context, drawable), IAztecAttributedSpan {
+                              editor: AztecText? = null) : AztecDynamicImageSpan(context, imageProvider), IAztecAttributedSpan {
 
     abstract val TAG: String
 
     private val overlays: ArrayList<Pair<Drawable?, Int>> = ArrayList()
 
+    private var innerPlaceholder: Bitmap? = null
+    private var isDirty = false
+    private val EXTRA_LOADING_SIZE = 500
```

Comment:
What's about increasing this to 1000?

Reply:
That would mean 2000px + the size of the edit text worth of images will be loaded. Can the most low-end devices handle it? cc: @mzorz